BaragonKubernetesIntegration/src/main/java/com/hubspot/baragon/kubernetes/KubernetesWatcher.java

```diff
@@ -0,0 +1,69 @@
+package com.hubspot.baragon.kubernetes;
+
+import java.util.Map;
+import java.util.Optional;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.google.inject.Inject;
+import com.google.inject.name.Named;
+import com.hubspot.baragon.kubernetes.listeners.KubernetesEventListener;
+
+import io.fabric8.kubernetes.api.model.Event;
+import io.fabric8.kubernetes.client.DefaultKubernetesClient;
+import io.fabric8.kubernetes.client.KubernetesClient;
+import io.fabric8.kubernetes.client.KubernetesClientException;
+import io.fabric8.kubernetes.client.Watcher;
+
+public class KubernetesWatcher implements Watcher<Event> {
+  private static final Logger LOG = LoggerFactory.getLogger(KubernetesWatcher.class);
+
+  private final KubernetesClient client;
+  private final ExecutorService executorService;
+  private final Map<KubernetesEvent, KubernetesEventListener> eventListeners;
+
+  @Inject
+  KubernetesWatcher(@Named(BaragonKubernetesIntegrationModule.EVENT_LISTENERS) Map<KubernetesEvent, KubernetesEventListener> eventListeners) {
+    this.client = new DefaultKubernetesClient();
```

Comment:
will eventually need a way to auth here. Default client works if env vars or set or if running in k8s, but we will likely need a way to specify a token